pyinstrument/stack_sampler.py

```diff
@@ -24,7 +25,15 @@
 
 thread_locals = threading.local()
 
-StackSamplerSubscriberTarget = Callable[[List[str], float, Optional["AsyncState"]], None]
+SubscriberCallstackFn = Callable[[List[str], float, Optional["AsyncState"]], None]
+SubscriberEventFn = Callable[[str, str, float], None]
+
+
+@dataclass
+class StackSamplerSubscriberTarget:
+    call_stack: SubscriberCallstackFn
+    event: SubscriberEventFn
```

Comment:
I wonder if we can combine this event and async_state change concepts into one interface. It seems that they're pretty similar to me.